### ISSUE TITLE
About gating_top_n

### ISSUE THREAD

**Issue by Heihaierr** [lucidrains/st-moe-pytorch#3]
Hi, I notice there is experiment with `top_n=1` in the paper of `st-moe`. But in `st_moe_pytorch.py`, 
`assert top_n >= 2, 'must be 2 or more experts'`
Can `top_n=1` work in this implementation?

**Comment by Heihaierr** [comment]
> @Heihaierr not here, as it was just to be faithful to the paper, which explored 2 and then a generalization of top-n (iirc) up to 3 and 4
> 
> i thought that top 1 didn't work that well?

Yes, but the paper also explored top-1 routing and shows improvement.

**Comment by lucidrains** [comment]
ah I see, yeah they did, but 2 is still recommended

![Screenshot_20231203-183548_Adobe Acrobat](https://github.com/lucidrains/st-moe-pytorch/assets/108653/195e4839-b94c-4d49-aa57-f69de4c38709)

**Comment by Heihaierr** [comment]
get it, thanks for quick reply

**Comment by moon4869** [comment]
If `top_n=1`, how should we achieve it?
